content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,166 @@
+are mergeable or not hence, the pull request is mergebale or not can also be
+decided by the payload of webhook also.
+
+=== How to run the Plugin
+
+See link:https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+and set credentials, owner and repository on your own and you will be good to go.
+
+=== What is avalilable now
+
+. Users are able to select to use Jenkinsfile.yaml for build.
+. Git Push step
+. harvest results and reports (and post in the PR)
+[loweralpha]
+.. `junit()`
+.. `findbugs()`
+.. `archiveArtifacts()`
+. Basic interface to parse and get build specifications from YAML file.
+
+=== Things decided
+
+. To build the plugin on the top of multibranch pipeline plugin. As that plugin has implementation of
+[loweralpha]
+.. Nice interface to show diffrent branch and pull requests build seperately with use of suitable plugins like Github, Bitbucket.
+.. Detect trusted revisions in a repository.
+.. Publishing of build status to the repository.
+. Convert the YAML configuration to declarative pipeline.
+. User will provide path to the script relative to the root directory of the repository
+  without extension (.sh or .bat) in the YAML file. The plugin will generate pipeline script to detect the
+  platform and call .sh or .bat script.
++
+  Example:
+    Path provided: ./scripts/hello
+    a. On UNIX machine “./scripts/hello.sh” will be called
+    b. On non-UNIX machine “./scripts/hello.bat” will be called.
+
+=== Implementations till now
+
+A first prototype of the plugin is ready. It supports all features of Multi-Branch Pipeline and offers the following features.
+
+Build description is defined via YAML file stored within the SCM repo. This plugin
+will depend on GitHub plugin, Bitbucket plugin, Gitlab plugin if users will be
+using respective paltfroms for their repositories.
+
+. Basic conversion of YAML to Declarative Pipeline: A class YamlToPipeline
+is written which will load the "Jenkinsfile.yaml" and make use of PipelineSnippetGenerator class
+to generate Declarative pipeline code.
+. Reporting of results.
+. Plugin is using Yaml from target branch right now. (Maybe this needs some discussion, example: 
+what if PR contains changes in Jenkinsfile.yaml)
+. Git Push step: To push the changes of PR to the target branch. This is implemented  
+using git-plugin, PushCommand is used for this from git-plugin. credentialId,
+branch name and repository url for intracting with Github, Bitbucket, etc
+will be taken automatically from respective plugins plugin. (Yes users have to configure in that way, you can see
+link:https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo])
+
+=== Jenkinsfile.yaml example
+
+Write now no format has been decided for the yaml file. But for this demo I am using below Jenkinsfile.yaml
+
+image:/images/post-images/gsoc-simple-pull-request-plugin/YAML.png[title="Jenkinsfile.yaml", role="center"]
```

Comment:
I suggest reverting this change and adding code instead so that users can copy-paste it. My proposal was to actually show some screenshots of resulting Pipeline in BlueOcean as we discussed on the call